src/lib/adapters/spark.js

```diff
@@ -54,20 +63,20 @@ SparkAdapter.prototype.postData = function(data) {
     formatted_message = split_message.pretext || split_message.text;
   }
 
-  self.robot.messageRoom.call(self.robot, recipient, formatted_message);
+  self.robot.messageRoom.call(self.robot, envelope, formatted_message);
```

Comment:
With this change, the `recipient` variable is no longer used anywhere, and should be removed, including on lines 53 through 55.

However, you may want to support the use of [`extra` feature](https://docs.stackstorm.com/chatops/aliases.html#testing-and-extending-alias-parameters) for this chat provider, so consider pulling attributes from `data.extra`.

Reply:
I have removed `recipient` and added `extra` attribute to the `envelope`